### ISSUE TITLE
[Bug]: checked-requires-onchange-or-readonly options are inverted

### ISSUE THREAD

**Issue by EvgenyOrekhov** [jsx-eslint/eslint-plugin-react#3711]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

- `"ignoreMissingProperties": true` works like `false`, and vice-versa
- `"ignoreExclusiveCheckedAttribute": true` works like `false`, and vice-versa

```json
{
    "react/checked-requires-onchange-or-readonly": [
      "error",
      {
        "ignoreMissingProperties": true,
        "ignoreExclusiveCheckedAttribute": true
      }
    ],
}
```

```js
function My() {
  return (
    <>
      <input type="checkbox" checked />
      <input type="checkbox" checked defaultChecked />
      <input type="radio" checked defaultChecked />
    </>
  );
}
```

The rule gives warnings for the above code.

eslint .

### Expected Behavior

- The rule SHOULD NOT give warnings when options are set to `true`
- The rule SHOULD give warnings when options are set to `false`
- `false` should be the new default value for the options (to warn by default, as it is now)

### eslint-plugin-react version

v7.34.0

### eslint version

v8.57.0

### node version

v20.11.1

**Comment by ljharb** [comment]
lol that seems like such an obvious bug that i'm not sure i even believe it, but indeed the tests seem to indicate this. cc @jaesoekjjang

**Comment by ljharb** [comment]
hmm, actually - the tests specifically cover it, correctly:
```js
    {
      code: '<input type="checkbox" checked />',
      options: [{ ignoreMissingProperties: true }],
    },
    {
      code: '<input type="checkbox" checked={true} />',
      options: [{ ignoreMissingProperties: true }],
    },
    {
      code: '<input type="checkbox" onChange={noop} checked defaultChecked />',
      options: [{ ignoreExclusiveCheckedAttribute: true }],
    },
    {
      code: '<input type="checkbox" onChange={noop} checked={true} defaultChecked />',
      options: [{ ignoreExclusiveCheckedAttribute: true }],
    },
```

also, the defaults are that both are `true`.

**Comment by ljharb** [comment]
Given that the tests seem to contradict your OP, can you provide a repro repo?

**Comment by EvgenyOrekhov** [comment]
@ljharb Where do you see those tests?

I see this:

https://github.com/jsx-eslint/eslint-plugin-react/blob/da1013c6760a997dca3050a4d1d8452f783584f1/tests/lib/rules/checked-requires-onchange-or-readonly.js#L41-L56

**Comment by jaesoekjjang** [comment]
OMG... I named the options in reverse. What a mistake... my bad🥲
How about renaming those options to 'noMissingProperties', 'noExclusiveCheckedAttrbute'?

**Comment by ljharb** [comment]
Renaming them seems more awkward than fixing the logic.

**Comment by jaesoekjjang** [comment]
> Renaming them seems more awkward than fixing the logic.

Yeah.. negative naming will make more confusion.
Opened new PR https://github.com/jsx-eslint/eslint-plugin-react/pull/3715
